mocktracer/propagation.go

```diff
@@ -52,7 +55,11 @@ func (t *TextMapPropagator) Inject(spanContext MockSpanContext, carrier interfac
 	writer.Set(mockTextMapIdsPrefix+"sampled", fmt.Sprint(spanContext.Sampled))
 	// Baggage:
 	for baggageKey, baggageVal := range spanContext.Baggage {
-		writer.Set(mockTextMapBaggagePrefix+baggageKey, baggageVal)
+		safeVal := baggageVal
+		if t.HTTPHeaders {
+			safeVal = url.QueryEscape(baggageVal)
+		}
+		writer.Set(mockTextMapBaggagePrefix+baggageKey, safeVal)
```

Comment:
agreed, but one thing at a time. The fact that `HTTPHeadersCarrier` was applying url-encoding was an actual bug in the API.